src/components/Profile.vue

```diff
@@ -0,0 +1,173 @@
+        <label>
+          <input type="checkbox" value="PersonalStatement" v-model="user.difficultCollegeProcess">
+          Personal Statement / Essay
+        </label>
+        <label>
+          <input type="checkbox" value="Resume" v-model="user.difficultCollegeProcess">
+          Resume
+        </label>
+        <label>
+          <input type="checkbox" value="Application" v-model="user.difficultCollegeProcess">
+          Filling out application
+        </label>
+        <label>
+          <input type="checkbox" value="Exams" v-model="user.difficultCollegeProcess">
+          SAT / ACT Exams
+        </label>
+        <label>
+          <input type="checkbox" value="ChoosingSchools" v-model="user.difficultCollegeProcess">
+          Choosing which schools to apply to
+        </label>
+      </div>
+      <ul v-show="'difficultCollegeProcess' != activeEdit" v-for="item in user.difficultCollegeProcess">
+        <li>{{ item }}</li>
+      </ul>
+      <button @click="editField('difficultCollegeProcess')">{{ fields['difficultCollegeProcess'] }}</button>
+    </div>
+
+    <div class="hasGuidanceCounselor">
+      <span>Have a guidance counselor:</span>
+      <select class="form-control" v-model="user.hasGuidanceCounselor" v-show="'hasGuidanceCounselor' == activeEdit">
+        <option></option>
+        <option>Yes</option>
+        <option>No</option>
+        <option>I don't know</option>
+      </select>
+      <span v-show="'hasGuidanceCounselor' != activeEdit">{{ user.hasGuidanceCounselor }}</span>
+      <button @click="editField('hasGuidanceCounselor')">{{ fields['hasGuidanceCounselor'] }}</button>
+    </div>
+
+    <div class="gpa">
+      <span>GPA:</span>
+      <span v-show="'gpa' != activeEdit">{{ user.gpa }}</span>
+      <input type="text" v-model="user.gpa" v-show="'gpa' == activeEdit">
+      <button @click="editField('gpa')">{{ fields['gpa'] }}</button>
+    </div>
+
+    <div class="collegeApplicationsText">
+      <span>List all colleges and universities you are considering apply to:</span>
+      <span v-show="'collegeApplicationsText' != activeEdit">{{ user.collegeApplicationsText }}</span>
+      <input type="text" v-model="user.collegeApplicationsText" v-show="'collegeApplicationsText' == activeEdit">
+      <button @click="editField('collegeApplicationsText')">{{ fields['collegeApplicationsText'] }}</button>
+    </div>
+  </div>
+
+  <button @click="saveProfile()">{{ saveBtnMsg }}</button>
+</div>
+
+</template>
+
+<script>
+import UserService from 'src/services/UserService'
+
+export default {
+  data() {
+    var user = UserService.getUser();
+    var fieldnames = ['firstname', 'lastname', 'highschool', 'currentGrade',
+    'expectedGraduation', 'difficultAcademicSubject', 'difficultCollegeProcess',
+    'hasGuidanceCounselor', 'gpa', 'collegeApplicationsText'];
+    var fields = [];
+    fieldnames.map(function(field) {
+      fields[field] = 'Edit';
```

Comment:
Interesting logic but it is non-obvious that this will be a button name. It would be clearer to either rename the state name to `fieldButtons` or similar, or simply use an object map of `true`/`false` values to indicate when a field is being edited (e.g., `field.gpa = true` onClick) and then doing `{{ field.gpa ? 'Done' : 'Edit' }}` or similar in the component, for each button content